### DISCUSSION TITLE
"Catch-all" blocks/failsafes for .qss files

### DISCUSSION THREAD

**Discussion by AzumaHazuki** [lxqt/lxqt#2345] [Q&A]
I've noticed that setting

```
QWidget {
     background: $SOME_COLOR;
     color: $SOME_OTHER_COLOR;
     /* etc etc etc*/
}
```

seems to apply properties to **all** the widgets in an LXQt theme...regardless of which .qss file this is placed in (i.e., a single instance, preferably in lxqt-panel.qss since that's where most of the theming is done, is enough).

Should we be making it advisory, or even mandatory, to head up lxqt-panel.qss files with a "globals" section to act as a catch-all for unthemed widgets? This might be good for purposes of theming consistency and allow developers to catch anything they missed while writing the individual .qss stylesheets out.

Edit: Maybe this would be a good code block for the start of an lxqt-panel.qss file?

```
/* Global "Catch-All" Settings */
QWidget {
      color:      palette(text);
      background:      palette(window);
      selection-color:      palette(highlighted-text);
      selection-background-color:      palette(highlight);
}
```
Of course this is for a theme intended to inherit colors from a palette; values that should stay static can just be hardcoded, i.e., specify the hex triplet or rgba(R, G, B, A) or whichever instead :)

**Comment by stefonarch** [comment]
Cannot reproduce your finding. Tested 
```
 QWidget {
     background: red;
     color: green;

} 

```

in runner.qss and panel.qss of ambiance, doesn't affect other. Did you restart the modules after change?

**Reply by tsujan** [reply]
No block in one QSS file can affect the others. As @stefonarch mentioned, restarting the Panel from LXQt Session Settings may be needed.

However, when you make a block like that in one QSS files, it'll affect all widgets below it, unless those widgets set different values. That's how stylesheets work anywhere.

**Reply by AzumaHazuki** [reply]
@tsujan, okay then...if there were a file, say "globals.qss" that just had that one block of defaults, would *that* act as a catch-all? It makes sense that files can't clobber one another (how the heck would they determine precedence, anyway?) but would having that QWidget{ (stuff;) } block in its own theme work as a default case?

**Reply by tsujan** [reply]
No. Each theme has its own stylesheets.

**Reply by AzumaHazuki** [reply]
I don't mean across themes, rather internal to each theme; the block in globals.css would act as something like the final statement in an old C-style switch/case block, so if an author forgot to theme a given LXQt widget in their theme (and _where_ is the list of all widgets and their states we can theme...?) this would provide some consistency.
